### ISSUE TITLE
Support for PDF standard A-3b

### ISSUE THREAD

**Issue by muratovic-ajdin81** [dompdf/dompdf#3534]
Hello,

After investigation I can see that dompdf
 cannot produce the PDF documents which are in standard A-3. Can we get support for this please?

**Comment by wblessen** [comment]
What exactly is missing?

**Comment by muratovic-ajdin81** [comment]
Produced PDF files are in standard A-1b

> What exactly is missing?

Produced PDF files are in standard A-1b and does not satisfy some new requirements for example Zugferd which starts from 2025. You can check the standard at https://demo.verapdf.org with one of your PDF files produced by dompdf, it is not A-3

**Comment by wblessen** [comment]
Which Requirements are exactly missing?

1. Embedding Fonts?
2. Color Profiles
3. Embedding Files

Anything else?

**Comment by muratovic-ajdin81** [comment]
> https://github.com/horstoeko/zugferd

Unfortunately https://github.com/horstoeko/zugferd does not work with PDF files produced by dompdf library. That's the whole point. To be able to embed anything into PDF file, the standard of PDF file must A-3. Dompdf library can produce only A-1 standard PDF files. So, https://github.com/horstoeko/zugferd does not work if pdf is produced by dompdf and that's my questions for - can we get support from dompdf to produce A-3 standard PDF files.

**Comment by wblessen** [comment]
Seems we have the same job :-)

Unfortunatly, I am also just a user of dompdf for many years, but have no clue, what it does under the hood.

So still, that someone can implement Standard A-3b, we first need a list of features, that have  to be implemented.

Can you help with this?

**Comment by mokorana** [comment]
Would be really nice to have that functionality directly in `dompdf `. However, [horstoeko](https://github.com/horstoeko/zugferd) solved the problem in his library by using `fpdf` to merge the PDF_without_XML and the Zugferd_XML into a new PDF which is in the format `PDF/A - 3b`.

Works like a charm.

**Comment by muratovic-ajdin81** [comment]
Herstoko did not solve the PDF A-3 problem because Herstoko does not produce PDF file by itself. Herstoko just use  existing A-3 PDF to attach XML file into it. So, you must have another library for PDF, for example FPDF. But, here the question is not about FPDF but about fompdf because of one simple reason - someone who use dompdf gor years now should re-do the whole code to make it working with FPDF. Should we mention how logic of pdf creation between these libraries is different in core?

**Comment by mokorana** [comment]
Maybe you should read this [class](https://github.com/horstoeko/zugferd/blob/master/src/ZugferdDocumentPdfMerger.php) a bit more carefully.

**Comment by muratovic-ajdin81** [comment]
It exactly the class with functionality that I described. Or you see something in it that can Create the PDF file?

**Comment by mokorana** [comment]
Okay, one more try from my side. Have a look at that [class](https://github.com/horstoeko/zugferd/blob/1140e93a41d6f8189ff0dd83819d0936a8cd3e13/src/ZugferdDocumentPdfBuilderAbstract.php) there are functions for generating and saving Document. Which do exactly what I described in my first reply.

**Comment by bsweeney** [comment]
There is an existing request to support Factur-X (#3443), a pull request to add some level of support for PDF/A (3269), a request to improve accessibility (#2774), among others. I have not had a chance to look into what we would need to do to support the standard as of yet though it looks like this particular feature requires reprioritizing based on upcoming requirements.

**Comment by muratovic-ajdin81** [comment]
> Okay, one more try from my side. Have a look at that [class](https://github.com/horstoeko/zugferd/blob/1140e93a41d6f8189ff0dd83819d0936a8cd3e13/src/ZugferdDocumentPdfBuilderAbstract.php) there are functions for generating and saving Document. Which do exactly what I described in my first reply.

Your point regarding the conversion is right but not if we talk about pdf files produced by dompdf library. The problem defined by opening this support request is that Application X already use dompdf for years and a very complex code is behind it. So, in such situation we cannot use the class ZugferdDocumentPdfBuilderAbstract - it will not work. Why? The reason is that pdf files created by dompdf does not satisfy the requirements of A-3b standard by its structure (for example one of issues is embedded fonts, etc.)
In summary, if we use dompdf to produce the pdf file the result is that such file cannot be used by horstoeko nor ZugferdDocumentPdfBuilderAbstract class to convert it to A-3b.
The best solution: get the support from dompdf to make it A-3b compatible
The worst solution: switch to FPDF library, re-do whole code for PDF file creation, use horstoeko\zugferd library to attach the xml file

**Comment by wblessen** [comment]
I try to meet the requirements from VeraPdf.

Validation Report
File:	test.pdf
Validation Profile:	PDF/A-3B validation profile
Compliance:	Failed

After setting a font and a MetaInformation, the following Rules are not satified:

1. https://github.com/veraPDF/veraPDF-validation-profiles/wiki/PDFA-Parts-2-and-3-rules#rule-6243-2
2. https://github.com/veraPDF/veraPDF-validation-profiles/wiki/PDFA-Parts-2-and-3-rules#rule-612-2
3. https://github.com/veraPDF/veraPDF-validation-profiles/wiki/PDFA-Parts-2-and-3-rules#rule-622-2
4. https://github.com/veraPDF/veraPDF-validation-profiles/wiki/PDFA-Parts-2-and-3-rules#rule-621141-1
5. https://github.com/veraPDF/veraPDF-validation-profiles/wiki/PDFA-Parts-2-and-3-rules#rule-6621-1

I am trying to solve them one by one, but any help is welcome.

**Comment by wblessen** [comment]
The solution is already implemented, just needs to be integrated.
See

https://github.com/dompdf/dompdf/compare/master...williamdes:dompdf:pdfa-2.0

**Comment by williamdes** [comment]
> The solution is already implemented, just needs to be integrated. See
> 
> [master...williamdes:dompdf:pdfa-2.0](https://github.com/dompdf/dompdf/compare/master...williamdes:dompdf:pdfa-2.0)

It is the PR https://github.com/dompdf/dompdf/pull/3269 but based on a different branch

**Comment by williamdes** [comment]
PDF/A is now released: https://github.com/dompdf/dompdf/releases/tag/v3.1.0
I am trying to figure out how to use it

**Comment by bsweeney** [comment]
FYI documentation is here: https://github.com/dompdf/dompdf/wiki/PDFA-Support
